main.go

```diff
@@ -1,133 +1,151 @@
-package main
-
-import (
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"log"
-	"net/http"
-	"sync"
-	"strconv"
-
-	"github.com/gorilla/mux"
-)
-
-type autoInc struct {
-    sync.Mutex
-    id int
-}
-
-func (a *autoInc) ID() (id int) {
-    a.Lock()
-    defer a.Unlock()
-
-    id = a.id
-    a.id++
-    return
-}

```

Comment:
Personally, I hate named returns. Some folks in Go like it - but it seems less clear to me.

Reply:
Yeah, I shouldnt have done this. That's a bad look